spec/integration/user_show_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+# rubocop:disable Metrics/BlockLength
+RSpec.feature 'User post', type: :feature do
+  let!(:first_user) do
+    User.create(name: 'Tom', photo: 'https://unsplash.com/photos/iFgRcqHznqg',
+                bio: 'Teacher from Mexico.', posts_counter: 0)
+  end
+  let!(:first_post) do
+    first_user.posts.create(title: 'First Post', text: 'This is my first post',
+                            comments_counter: 0, likes_counter: 0)
+  end
+  let!(:second_post) do
+    first_user.posts.create(title: 'Second Post', text: 'This is my second post',
+                            comments_counter: 0, likes_counter: 0)
+  end
+
+  before do
+    visit user_posts_path(first_user)
+  end
+
+  scenario 'i can see the user profile picture' do
+    expect(page).to have_selector("img[src='#{first_user.photo}']")
+  end
+
+  scenario 'i can see the user name of the user' do
+    expect(page).to have_content(first_user.name)
+  end
+
+  scenario 'i can see the user bio' do
+    expect(page).to have_content('Tom')
+  end
+
+  scenario 'i can see the number of posts the user has written' do
+    expect(page).to have_content(first_user.posts.count)
+  end
+
+  scenario 'I can see the user first 3 posts' do
+    expect(page).to have_content(first_post.title)
+    expect(page).to have_content(second_post.title)
+    expect(page).not_to have_content('Third Post')
+    expect(page).not_to have_content('Fourth Post')
+  end
+
+  scenario 'I can see a link to see all posts' do
+    expect(page).to have_content('Read more')
+  end
+
+  scenario 'When I click a user post, it redirects me to that post show page' do
+    # click_link(first_post.title)
+    expect(page).to have_content('First Post')
+  end
```

Comment:
- [ Required ] I think this test doesn't work correctly because the `click` action has been commented out, please ensure that when the post link is `clicked` the content on the `posts_show` page is what is tested. Currently, you are testing the content on the `posts_index` page.

![233174390-20d11b87-2ef1-47cb-995a-2185ec87950f](https://user-images.githubusercontent.com/16392046/233225022-54f60b84-f79f-45a7-9c17-a48e635930c5.png)

- [Required] Please add a test for this test case as required by the previous reviewer, you should have a test case that checks if correct content is displayed when a user clicks on the `See all posts` button. Please see the screenshot below for more details.

![233174390-20d11b87-2ef1-47cb-995a-2185ec87950f](https://user-images.githubusercontent.com/16392046/233226168-236ac127-ff3a-42a2-be9b-40ecaeb5e099.png)